server/hotrod/src/main/java/org/infinispan/server/hotrod/ClientListenerRegistry.java

```diff
@@ -333,7 +341,7 @@ boolean isSendEvent(CacheEntryEvent<byte[], byte[]> event) {
       void init() {
          ch.closeFuture().addListener(f -> {
             log.debug("Channel disconnected, removing event sender listener for id: " + Util.printArray(listenerId));
-            cache.removeListenerAsync(this)
+            removeClientListener(listenerId, cache)
```

Comment:
@wburns I noticed that `unblockCommands()` is called when the listener is removed because the channel was closed by the client, but not when the client removes the listener explicitly. Is that intentional?

Reply:
Are the failures connected to this ?

Reply:
No @tristantarrant, but they were related to the pull request: I changed the authentication tests to reuse the Hot Rod server, but I forgot I didn't make them reuse the RemoteCacheManager.